evalml/tests/component_tests/test_arima_regressor.py

```diff
@@ -0,0 +1,151 @@
+    y_pred_a = a_clf.predict(params=(1, 0, 0))
+
+    y = y.reset_index(drop=True)
+    assert isinstance(X.index, pd.DatetimeIndex)
+    assert not isinstance(y.index, pd.DatetimeIndex)
+
+    clf = ARIMARegressor(p=1, d=0, q=0)
+    clf.fit(X=X, y=y)
+    y_pred = clf.predict(X=X, y=y)
+
+    assert (y_pred == y_pred_a).all()
+
+
+def test_fit_predict_ts_with_ynotX_index(ts_data):
+    X, y = ts_data
+
+    a_clf = arima.ARIMA(endog=y, exog=X, order=(1, 0, 0), dates=X.index)
+    a_clf.fit(solver='nm')
+    y_pred_a = a_clf.predict(params=(1, 0, 0))
+
+    X_no_ind = X.reset_index(drop=True)
+    assert isinstance(y.index, pd.DatetimeIndex)
+    assert not isinstance(X_no_ind.index, pd.DatetimeIndex)
+
+    clf = ARIMARegressor(p=1, d=0, q=0)
+    clf.fit(X=X_no_ind, y=y)
+    y_pred = clf.predict(X=X, y=y)
+
+    assert (y_pred == y_pred_a).all()
+
+
+def test_predict_ts_without_X(ts_data):
+    X, y = ts_data
+
+    a_clf = arima.ARIMA(endog=y, exog=X, order=(1, 0, 0), dates=X.index)
+    a_clf.fit(solver='nm')
+    y_pred_a = a_clf.predict(params=(1, 0, 0))
+
+    clf = ARIMARegressor(p=1, d=0, q=0)
+    clf.fit(X=X, y=y)
+    y_pred = clf.predict(y=y)
+
+    assert (y_pred == y_pred_a).all()
+
+
+def test_fit_ts_with_notXnoty_index(ts_data):
+    X, y = ts_data
+    X = X.reset_index(drop=True)
+    y = y.reset_index(drop=True)
+    assert not isinstance(y.index, pd.DatetimeIndex)
+    assert not isinstance(X.index, pd.DatetimeIndex)
+
+    clf = ARIMARegressor(p=1, d=0, q=0)
+    with pytest.raises(ValueError, match="If not it will look for the datetime column in the index of X or y."):
+        clf.fit(X=X, y=y)
+
+
+def test_predict_ts_with_notX_index(ts_data):
+    X, y = ts_data
+    X = X.reset_index(drop=True)
+    assert not isinstance(X.index, pd.DatetimeIndex)
+
+    clf = ARIMARegressor(p=1, d=0, q=0)
+    clf.fit(X=X, y=y)
+    with pytest.raises(ValueError, match="If not it will look for the datetime column in the index of X."):
+        clf.predict(X)
+
+
+def test_fit_ts_without_y(ts_data):
+    X, y = ts_data
+
+    clf = ARIMARegressor(p=1, d=0, q=0)
+    with pytest.raises(ValueError, match="ARIMA Regressor requires y as input."):
+        clf.fit(X=X)
+
+
+def test_fit_predict_ts_no_X(ts_data):
+    X, y = ts_data
+
+    a_clf = arima.ARIMA(endog=y, order=(1, 0, 0), dates=y.index)
+    a_clf.fit(solver='nm')
+    y_pred_a = a_clf.predict(params=(1, 0, 0))
+
+    clf = ARIMARegressor(p=1, d=0, q=0)
+    clf.fit(X=None, y=y)
+    y_pred = clf.predict(X=X, y=y)
+
+    assert (y_pred == y_pred_a).all()
+
+
+def test_fit_predict_date_col_named(ts_data):
+    X, y = ts_data
+
+    a_clf = arima.ARIMA(endog=y, exog=X, order=(1, 0, 0), dates=X.index)
+    a_clf.fit(solver='nm')
+    y_pred_a = a_clf.predict(params=(1, 0, 0))
+
+    X = X.reset_index()
+    clf = ARIMARegressor(p=1, d=0, q=0, date_column='index')
+    clf.fit(X=X, y=y)
+    y_pred = clf.predict(X=X, y=y)
+
+    assert (y_pred == y_pred_a).all()
```

Comment:
If you decide to leave `get_dates_fit` and `get_dates_predict` as public methods, can you add some tests for that?

Reply:
Isn't this functionality (looking at a datatable/frame of features and determining date time columns) used somewhere else?  If you go the route of public methods, would definitely like an issue to refactor those other estimators or whatnot that use that logic.  @freddyaboulton do you use this in time series?